packages/fs-detectors/src/workspaces/get-workspace-package-paths.ts

```diff
@@ -113,3 +124,21 @@ async function getPnpmWorkspacePackagePaths({
 
   return getPackagePaths(packages, fs);
 }
+
+async function getRushWorkspacePackagePaths({
+  fs,
+}: GetPackagePathOptions): Promise<string[]> {
+  const rushWorkspaceAsBuffer = await fs.readFile('rush.json');
+
+  const { projects = [] } = JSON.parse(
+    rushWorkspaceAsBuffer.toString()
+  ) as RushWorkspaces;
+
+  if (Array.isArray(projects)) {
```

Comment:
^^ Same as above, would it be acceptable to return the empty array but to also log the error? Or do you think we should throw an error

Reply:
Depends if we think there are cases where `rush.json` is not a corresponding Rush workspace (like a user just happened to name it that).

If we know its always a Rush workspace, then we should show the error to the user. If not we should just consume it and move onto the next file

Reply:
It would at this point that we know they have a `rush.json` which means they should have rush workspaces. If `projects` is not an array that would mean the user has their `rush.json` configured incorrectly (this should be rare because most cases it shows the user errors for the format so if they truly are using Rush they wouldn't be able to run it with an incorrectly formatted `rush.json`

Reply:
> at this point that we know they have a rush.json which means they should have rush workspaces.

Why is that? Couldn't a user just have an arbitrary file named `rush.json` and store whatever json they want in there for their own purpose?

This is different than `yarn.lock` which is a very established filename that is unlikely to collide with user's files.

Reply:
I think it's reasonable to look only for `projects: [{ projectFolder:...}]` for now, but I'd like to iterate on schema validation for better error messages.

Given that this file is not a common standard where we can assume it's a Rush config file, I think we should debug log any errors we find here. This would give someone the ability to figure out why their `rush.json` didn't work properly when they were using Rush. It would also hide these details completely when they were not using Rush.

Reply:
@EndangeredMassa For `vercel` where do the logs go? Would `console.error` go to the customer logs?

Reply:
Hrm. Because we're in `fs-detectors`, I don't think there's a clear way to log this. Ideally, `getWorkspacePackagePaths` would return a complicated response that includes what workspace package managers it tried to parse and what the results were. Then the API could log that itself. It could go further and return that more complicated response as some metadata in the API response, where the caller could log it if it made sense. That's way out of scope for this PR, though.

Maybe instead when we expand Vercel CLI to better support monorepos, we can use `fs-detectors` to parse workspace package manager configs and show validation failures. That would give the user a path to figuring out what's breaking.

For this PR, I think we're good to go.